Fix DeleteOptions on Public Ip Address (issue#38806) (#39096)
Azure/azure-sdk-for-java: sdk/resourcemanager/azure-resourcemanager-network/CHANGELOG.md

### Features Added

> - Supported set `DeleteOptions` on a public ip address in `NetworkInterface`, for allow deleting the public ip automatically when the vm is deleted.

Reviewer: IP

Reviewer: virtual machine (not vm)

Author: Fixed in the new version.

> - Supported set `DeleteOptions` on a public IP address in `NetworkInterface`, for allow deleting the public IP automatically when the virtual machine is deleted.

Reviewer 2: ```suggestion
- Supported setting `DeleteOptions` for public IP addresses associated with `NetworkInterface`.
```

Author: Fixed in the new version.
